taxonomy/utils.py

```diff
@@ -81,29 +83,39 @@ def get_product_skill_model_and_identifier(product_type):
     """
     Return the Skill Model along with its identifier based on product type.
     """
-    return (ProgramSkill, 'program_uuid') if product_type == ProductTypes.Program else (CourseSkills, 'course_key')
+    if product_type == ProductTypes.Program:
+        return (ProgramSkill, 'program_uuid')
+    if product_type == ProductTypes.XBlock:
+        return (XBlockSkills, 'usage_key')
+    if product_type == ProductTypes.XBlockThrough:
+        return (XBlockSkillThrough, 'xblock_id')
+    return (CourseSkills, 'course_key')
 
 
-def update_skills_data(key_or_uuid, skill_external_id, confidence, skill_data, product_type):
+def update_skills_data(key_or_uuid, skill_external_id, confidence, skill_data, product_type, **kwargs):
     """
-    Persist the skills data in the database either for Program or Course.
+    Persist the skills data in the database for Program, Course or XBlock.
     """
-    skill, __ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
+    skill, _ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
+    if product_type == ProductTypes.XBlock:
+        x_model, x_identifier = get_product_skill_model_and_identifier(product_type)
+        product_type = ProductTypes.XBlockThrough
+        xblock, _ = x_model.objects.update_or_create(
+            **{x_identifier: key_or_uuid},
+            defaults={"hash_content": kwargs.get("hash_content"), "auto_processed": True},
```

Comment:
Why are we doing two calls to update_or_create? Also, when do we check if the skill is blacklisted? I am probably not able to understand this.